### PR TITLE
Auth/PM-6198 - Registration with Email Verification - Add email clicked endpoint

### PR DESCRIPTION
**Pull request by JaredSnider-Bitwarden**
## 🎟️ Tracking

[PM-6198](https://bitwarden.atlassian.net/browse/PM-6198)

See clients PR: https://github.com/bitwarden/clients/pull/10139

## 📔 Objective

To add a new endpoint to the Identity `AccountsController` to submit a reference event when a user lands on the registration finish screen from clicking the emailed link. 

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-6198]: https://bitwarden.atlassian.net/browse/PM-6198?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Identity/Controllers/AccountsController.cs [hunk 1]
```diff
@@ -102,6 +105,39 @@ public async Task<IActionResult> PostRegisterSendVerificationEmail([FromBody] Re
...
+
+        await _referenceEventService.RaiseEventAsync(refEvent);
+
+        if (!tokenValid || userExists)
+        {
+            throw new BadRequestException("Expired link. Please restart registration or try logging in. You may already have an account");
+        }
+
+        return Ok();
+
```
Comment (ike-kottlowski) [inline src/Identity/Controllers/AccountsController.cs:137]: ⛏️ extra white space after `return`

**Review by ike-kottlowski — APPROVED** [review]
Looks good.

File: src/Core/Tools/Enums/ReferenceEventSource.cs [hunk 2]
```diff
@@ -10,6 +10,6 @@ public enum ReferenceEventSource
     User,
     [EnumMember(Value = "provider")]
     Provider,
-    [EnumMember(Value = "registrationStart")]
-    RegistrationStart,
+    [EnumMember(Value = "registration")]
+    Registration,
```
Comment (djsmith85) [inline src/Core/Tools/Enums/ReferenceEventSource.cs:14]: @Mailman14 Any issues with renaming this?

Comment (JaredSnider-Bitwarden) [reply src/Core/Tools/Enums/ReferenceEventSource.cs:14]: @djsmith85 , as info, there have not been any changes to the reference events repo yet to process the new registration reference events (I added the other event). That work is captured [here](https://bitwarden.atlassian.net/browse/PM-9372).

**Review by djsmith85 — APPROVED** [review]

**Comment by JaredSnider-Bitwarden** [comment]
Merging as this is behind a feature flag.

**Pull request merged**